MvvmCross/ViewModels/MvxAppStart.cs

```diff
@@ -3,32 +3,47 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Threading;
-using System.Threading.Tasks;
 using MvvmCross.Exceptions;
 using MvvmCross.Logging;
 using MvvmCross.Navigation;
 
 namespace MvvmCross.ViewModels
 {
-    public class MvxAppStart<TViewModel>
-        : IMvxAppStart
-        where TViewModel : IMvxViewModel
+    public abstract class MvxAppStart : IMvxAppStart
     {
-        protected readonly IMvxNavigationService NavigationService;
-
         private int startHasCommenced;
-
-        public MvxAppStart(IMvxNavigationService navigationService)
-        {
-            NavigationService = navigationService;
-        }
         
         public void Start(object hint = null)
         {
             // Check whether Start has commenced, and return if it has
             if (Interlocked.CompareExchange(ref startHasCommenced, 1, 0) == 1)
                 return;
 
+            Startup(hint);
+        }
+
+        protected abstract void Startup(object hint = null);
```

Comment:
This would break all apps that use it currently. Maybe make the first method Startup, and the abstract one Start so it stays the same?

Reply:
Indeed. But the breaking change was caused the Setup refactor... As the way AppStart is called has changed, there's now some necessary code to make it thread safe, and that code was only on the generic version of it. 

Same as the [StarWarsSample](https://github.com/MvvmCross/MvvmCross-Samples/pull/114/files#diff-56bd2bea371da509b91bfb9c38d72189R7), most complex apps do have a custom AppStart and therefore I think it's safer to break stuff and make developers inherit their custom AppStarts from `MvxAppStart` instead of `IMvxAppStart`

Regarding the naming... `Start` is now a "wrapper" call for `Startup`, which contains all the thread safe code verification. As the method is public and can be called from literally anywhere, and we want the code to be always thread safe, I made Startup `protected` and left Start public.